products/BellHybrid/apps/application-bell-settings/models/DateTimeUnitsModel.cpp

```diff
@@ -12,37 +13,48 @@
 #include <service-time/Constants.hpp>
 #include <service-time/api/TimeSettingsApi.hpp>
 #include <service-time/service-time/TimeMessage.hpp>
+#include <widgets/DateSetSpinner.hpp>
 #include <widgets/TimeSetFmtSpinner.hpp>
 
 #include <ctime>
 
 namespace app::bell_settings
 {
-    TimeUnitsModel::TimeUnitsModel(app::ApplicationCommon *app) : application(app)
+    namespace
+    {
+        constexpr auto factoryResetDate   = date::year(2022) / date::month(8) / date::day(18);
+        constexpr int factoryResetTime    = 12 * 60 * 60; // 12:00 PM
+        constexpr auto factoryRestTimeFmt = utils::time::Locale::TimeFormat::FormatTime12H;
+
+    } // namespace
+
+    DateTimeUnitsModel::DateTimeUnitsModel(app::ApplicationCommon *app) : application(app)
     {}
 
-    TimeUnitsModel::~TimeUnitsModel()
+    DateTimeUnitsModel::~DateTimeUnitsModel()
     {
         clearData();
     }
 
-    auto TimeUnitsModel::requestRecordsCount() -> unsigned int
+    auto DateTimeUnitsModel::requestRecordsCount() -> unsigned int
     {
         return internalData.size();
     }
 
-    auto TimeUnitsModel::getMinimalItemSpaceRequired() const -> unsigned int
+    auto DateTimeUnitsModel::getMinimalItemSpaceRequired() const -> unsigned int
     {
         return style::sidelistview::list_item::w;
     }
 
-    auto TimeUnitsModel::getItem(gui::Order order) -> gui::ListItem *
+    auto DateTimeUnitsModel::getItem(gui::Order order) -> gui::ListItem *
     {
         return getRecord(order);
     }
 
-    void TimeUnitsModel::createData()
+    void DateTimeUnitsModel::createData()
     {
+        dateSetListItem = new gui::DateSetListItem();
+        internalData.push_back(dateSetListItem);
```

Comment:
who has ownership of the underlying memory? if internalData, then consider using `std::move`

Reply:
dateSetListItem is used in another places in cpp file, using moved form object after move can cause undefined behaviour

Reply:
@SP2FET It's owned by `app::InternalModel<gui::ListItem *>`. They are deleted in the desctuctor of this base class.

I have a different but related question though. Why are raw pointers used instead of some smart pointers? If an exception is thrown by `push_back` then there will be a memory leak. Why not use `unique_ptr`? And here we're circling back to @SP2FET question. With `unique_ptr` the ownership would be clear. You either have to keep it as a member of `DateTimeUnitsModel` or move it to `internalData` and use it from there. Or you could use `shared_ptr` and the ownership would also be clear.

Why is lack of clear ownership a problem? Consider calling `DateTimeUnitsModel::clearData()` which calls `InternalModel::eraseInternalData()` which calls delete for all of the items. So at this point the items were deleted but the pointers inside `DateTimeUnitsModel` still are not `null`. So now we could try to call something from a place of memory which isn't ours. Etc. With this design you should probably have to remember to set these pointers to `null`, maybe add an `assert` somewhere to detect calling null pointers, etc. So basiclaly you have to care about them in two places at once. Smart pointers and/or clearly defined ownership would solve this automatically.

Reply:
The owner of  dateSetListItem is internalData. As we discussed it internally, memory exception would crash system anyway so its acceptable :P . Generally gui design is based on raw pointers and memory is managed manually. The destructor is called in base/parent/Item class

Reply:
Thanks for the explanation.